### PR TITLE
Translate CG transform to DIPs before applying it to D2D IWICBitmap render target

### LINKED ISSUE #1041: CTCatalog test app does not render any text on Phone
Seems like a regression but not sure what caused this.

### PR DESCRIPTION
**Pull request by mnithish**
Fixes #1041

### CODE REVIEW

**Comment by mnithish** [comment]
@ms-jihua is added to the review. #Closed

**Comment by mnithish** [comment]
@aballway is added to the review. #Closed

**Comment by ms-jihua** [comment]
:shipit:

File: Frameworks/CoreGraphics/CGContextCairo.mm [hunk 1]
```diff
@@ -2013,6 +2014,16 @@
...
     // Perform translation required to handle scaling.
     CGFloat verticalScalingFactor = sqrt((transform.b * transform.b) + (transform.d * transform.d));
+    // TODO::
+    // Issue #1057 - CG today talks in Pixels but DirectWrite and D2D only understand DIPs. Because CG talks in Pixels,
+    // curState->curTransform contains both the device scaling factor and any user scale applied to it causing the below D2D render target
+    // transform calculation to be way off on devices that have scaling. Workaround for now is to divide the verticalScalingFactor and
+    // height here with the device scaling factor so they get converted to DIPs and then apply the transform to the D2D render target.
+    // Though this workaround is sufficient to get text rendered on the user screen on all Windows form factor devices, this will not work
+    // when text is rendered on a PDF or printer.
+    float hostDisplayScale = static_cast<float>([[WGDDisplayInformation getForCurrentView] resolutionScale] / 100.0f);
```
Comment (rajsesh-msft) [inline Frameworks/CoreGraphics/CGContextCairo.mm]: can this be cached in the context initialization and used?

**Comment by rajsesh-msft** [comment]
:shipit:

File: Frameworks/CoreGraphics/CGContextCairo.mm (hunk 1 above)
Comment (mnithish) [inline Frameworks/CoreGraphics/CGContextCairo.mm]: I was about to update the CR for that when my VS stopped working for me :-|

---
In reply to: [81259574](https://github.com/Microsoft/WinObjC/pull/1058#discussion_r81259574) [](ancestors = 81259574)

Comment (mnithish) [inline Frameworks/CoreGraphics/CGContextCairo.mm]: Making it static instead as projections do not work in UT environment and causes CG UTs to throw

---
In reply to: [81263587](https://github.com/Microsoft/WinObjC/pull/1058#discussion_r81263587) [](ancestors = 81263587,81259574)

**Comment by aballway** [comment]
:shipit:

**Pull request merged**